src/model/resources/storage/vre_stor.jl

```diff
@@ -0,0 +1,279 @@
+in LICENSE.txt.  Users uncompressing this from an archive may not have
+received this license file.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+@doc raw"""
+	vre_stor(EP::Model, inputs::Dict, Reserves::Int, MinCapReq::Int, EnergyShareRequirement::Int, CapacityReserveMargin::Int)
+
+"""
+function vre_stor(EP::Model, inputs::Dict, Reserves::Int, MinCapReq::Int, EnergyShareRequirement::Int, CapacityReserveMargin::Int, StorageLosses::Int)
+
+	println("VRE-Storage Module")
+
+	dfGen_VRE_STOR = inputs["dfGen_VRE_STOR"]
+
+	T = inputs["T"]     # Number of time steps (hours)
+	Z = inputs["Z"]     # Number of zones
+	VRE_STOR = inputs["VRE_STOR"] 	# Number of generators
+    NEW_CAP_VRE_STOR = inputs["NEW_CAP_VRE_STOR"]
+    RET_CAP_VRE_STOR = inputs["RET_CAP_VRE_STOR"]
+    NEW_CAP_ENERGY_VRE_STOR = inputs["NEW_CAP_ENERGY_VRE_STOR"]
+    RET_CAP_ENERGY_VRE_STOR = inputs["RET_CAP_ENERGY_VRE_STOR"]
+    NEW_CAP_GRID = inputs["NEW_CAP_GRID"]
+    RET_CAP_GRID = inputs["RET_CAP_GRID"]
+
+    START_SUBPERIODS = inputs["START_SUBPERIODS"]
+	INTERIOR_SUBPERIODS = inputs["INTERIOR_SUBPERIODS"]
+	hours_per_subperiod = inputs["hours_per_subperiod"] #total number of hours per subperiod
+    
+    dfGen_VRE_STOR = inputs["dfGen_VRE_STOR"]
+
+    ### Variables ###
+
+    # Retired capacity of resource VRE from existing capacity [MW]
+    @variable(EP, vRETCAP_VRE[y=1:VRE_STOR] >= 0);
+    
+    # New installed capacity of resource VRE [MW]
+    @variable(EP, vCAP_VRE[y=1:VRE_STOR] >= 0);
+
+    # Retired grid capacity [MW]
+    @variable(EP, vRETGRIDCAP[y=1:VRE_STOR] >= 0)
+
+    # New installed grid capacity [AC MW]
+    @variable(EP, vGRIDCAP[y=1:VRE_STOR] >= 0)
+
+    # Energy storage reservoir capacity retired for VRE storage [MWh]
+    @variable(EP, vRETCAPSTORAGE_VRE_STOR[y=1:VRE_STOR] >= 0)	
+
+    # Energy storage reservoir capacity (MWh capacity) built for VRE storage [MWh]
+    @variable(EP, vCAPSTORAGE_VRE_STOR[y=1:VRE_STOR] >= 0)
+
+    # "Behind the inverter" generation [MW]
+    @variable(EP, vP_DC[y=1:VRE_STOR, t=1:T] >= 0)
+
+    # "Behind the inverter" charge from the VRE-component [MW]
+    @variable(EP, vCHARGE_DC[y=1:VRE_STOR, t=1:T] >= 0)
+
+    # "Behind the inverter" discharge [MW]
+    @variable(EP, vDISCHARGE_DC[y=1:VRE_STOR, t=1:T] >= 0)
+
+    # Energy injected into the grid by VRE_STOR at hour "t" [MW]
+    @variable(EP, vP_VRE_STOR[y=1:VRE_STOR,t=1:T] >=0);
+
+    # Energy withdrawn from grid by resource VRE_STOR at hour "t" [MW] 
+    @variable(EP, vCHARGE_VRE_STOR[y=1:VRE_STOR,t=1:T] >= 0);
+
+    # Storage level of resource "y" at hour "t" [MWh] on zone "z"
+    @variable(EP, vS_VRE_STOR[y=1:VRE_STOR,t=1:T] >= 0);
+
+	### Expressions ###
+
+    # Total capacity expressions for VRE-component, storage-component, and grid-component.
+	#@expression(EP, eTotalCap_VRE[y in 1:VRE_STOR], 
+    #    if y in intersect(NEW_CAP_VRE_STOR, RET_CAP_VRE_STOR) # Resources eligible for new capacity and retirements
```

Comment:
@aneeshamanocha can you remove the commented-out stuff here if it's not being used?